src/__experimental__/components/decimal/decimal.component.js

```diff
@@ -256,40 +159,23 @@ class Decimal extends React.Component {
    * Format a user defined value
    */
   formatValue = (value) => {
-    invariant(
-      !this.isNaN(value),
-      `The supplied decimal (${value}) is not a number`
-    );
+    if (this.isNaN(value)) {
+      return value;
+    }
     if (value === "") {
       return value;
     }
-    return I18nHelper.formatDecimal(value, this.getSafePrecisionProp());
-  };
-
-  /**
-   * Perform a string replacement on the input so we can see what the value will be after the change
-   * This allows us to prevent the user input if it is invalid
-   */
-  replace = (ev) => {
-    const {
-      target: { selectionStart, selectionEnd, value },
-      clipboardData,
-      type,
-    } = ev;
-    let { key: change } = ev;
-
-    if (type === "paste") {
-      change = clipboardData.getData("text/plain");
+    const { separator } = I18nHelper.format();
+    const parts = value.split(separator);
+    if (this.props.precision === 0) {
```

Comment:
If you enter `1234.56` it will be re-formatted as `1,234` and will loose precision. We don't want this to happen. Please add a test that covers this.

Reply:
Test has been added to cover this scenario.